subiquity/server/controllers/mirror.py

```diff
@@ -143,13 +162,39 @@ async def apply_autoinstall_config(self, context):
             log.debug("Skipping mirror check since network is not available.")
             return
 
-        try:
-            await self.try_mirror_checking_once()
-        except AptConfigCheckError:
-            log.debug("Retrying in 10 seconds...")
+        # Try each mirror one after another.
+        compatibles = self.model.compatible_primary_candidates()
+        for idx, candidate in enumerate(compatibles):
+            if idx != 0:
+                # Sleep before testing the next candidate..
+                log.debug("Will check next candiate mirror after 10 seconds.")
+                await asyncio.sleep(10)
```

Comment:
Can't this go at the end of the for loop?

Reply:
I would definitely prefer to but I also want to avoid sleeping after the final test, because it only makes sense to sleep for 10 seconds (which is a pretty bad penalty if there is still something to test after.

I came up with this enumerate + reversed logic construct to do exactly that, but I hate how this looks so any suggestion is appreciated :)

Basically, I want to sleep between iterations, rather than after each iteration. I guess this is somewhat similar to `'_'.join(("a", "b", "c"))` returning `"a_b_c"` and not `"a_b_c_"`.

Reply:
Yes but in this case the loop is usually exited via break?

Reply:
If a mirror is usable, yes. Otherwise, no.